src/posix/platform/netif.cpp

```diff
@@ -292,8 +292,8 @@ static void InitNetaskWithPrefixLength(struct in6_addr *address, uint8_t prefixL
 
 static uint8_t NetmaskToPrefixLength(const struct sockaddr_in6 *netmask)
 {
-    return otIp6PrefixMatch(static_cast<const otIp6Address *>(netmask->sin6_addr.s6_addr),
-                            static_cast<const otIp6Address *>(allOnes));
+    return otIp6PrefixMatch(reinterpret_cast<const otIp6Address *>(netmask->sin6_addr.s6_addr),
+                            reinterpret_cast<const otIp6Address *>(allOnes));
```

Comment:
👍

Reply:
Oops. I meant to push this in a separate PR :)